src/Altinn.App.Core/Features/Action/UserActionFactory.cs

```diff
@@ -21,14 +23,30 @@ public UserActionFactory(IEnumerable<IUserAction> actionHandlers)
     /// Find the implementation of <see cref="IUserAction"/> based on the actionId
     /// </summary>
     /// <param name="actionId">The id of the action to handle.</param>
-    /// <returns>The first implementation of <see cref="IUserAction"/> that matches the actionId. If no match <see cref="NullUserAction"/> is returned</returns>
-    public IUserAction GetActionHandler(string? actionId)
+    /// <returns>The first implementation of <see cref="IUserAction"/> that matches the actionId. If no match null is returned</returns>
+    public IUserAction? GetActionHandlerOrDefault(string? actionId)
+    {
+        if (actionId != null)
+        {
+            return _actionHandlers.FirstOrDefault(ah => ah.Id.Equals(actionId, StringComparison.OrdinalIgnoreCase));
+        }
+
+        return null;
```

Comment:
Noticed the next method where you can specify the defaultAction so I guess that should be used in cases where you would like to have the NullUserAction

Reply:
Opted to return null here so that the aciton controller could return 404 if the user is authenticated to perform an action with a given name, but there is no registered service with that id. Think that would make debugging a typo easier.
The process engine defaults to the NullUserAction as there might not be a need for custom code just an action to make a decision in a following gateway

Reply:
Noticed that the controller didn't return 404. Have added the code for that with a test